src/sdf/tape.h

```diff
@@ -0,0 +1,284 @@
+// Copyright 2024 The Manifold Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#pragma once
+
+#include <cmath>
+#include <cstdint>
+#include <cstdio>
+#include <cstring>
+#include <limits>
+
+#include "interval.h"
+#include "manifold/vec_view.h"
+
+namespace manifold::sdf {
+
+enum class OpCode : uint8_t {
+  NOP,
+  RETURN,
```

Comment:
Because SDFs typically doesn't have control flow, there is no control flow here as well. Complicated SDFs will require runtime optimization for fast execution and optimize away the unused path, which will be implemented later.